### PR TITLE
Added Authorization

### LINKED ISSUE #25: Add authorisation to all commands
All commands should be authorised.

E.g. Members requests should be authorised against the teams they belong too. I.e. you cannot request things in a team Slack channel (if it's public or you're invited) for a team that you are not a part of.

### PR DESCRIPTION
**Pull request by kbristow**
Added authorization to rest commands to ensure only users that are members of the teams owning resources can make changes to said resources.

Resolves #25

### CODE REVIEW

File: nucleus/src/main/java/za/co/absa/subatomic/domain/project/NewProject.java [hunk 1]
```diff
@@ -16,5 +19,7 @@
 
     private TeamMemberId createdBy;
 
-    private TeamId team;
```
Comment (donovanmuller) [inline nucleus/src/main/java/za/co/absa/subatomic/domain/project/NewProject.java]: Why the change away from `TeamId`?

File: nucleus/src/main/java/za/co/absa/subatomic/domain/team/Team.java [hunk 2]
```diff
@@ -70,6 +69,13 @@ void on(TeamCreated event) {
 
     @CommandHandler
     void when(AddTeamMembers command) {
+
```
Comment (donovanmuller) [inline nucleus/src/main/java/za/co/absa/subatomic/domain/team/Team.java]: Please remove blank line after method declaration.

File: nucleus/src/main/java/za/co/absa/subatomic/domain/application/Application.java [hunk 3]
```diff
@@ -33,6 +34,12 @@
 
     @CommandHandler
     public Application(NewApplication command) {
+        if (!command.getAllAssociateProjectOwnerAndMemberIds()
+                .contains(command.getRequestedBy().getTeamMemberId())) {
+            throw new SecurityException(
+                    "requestedBy member is not a valid member of any team associated to the owning project.");
```
Comment (donovanmuller) [inline nucleus/src/main/java/za/co/absa/subatomic/domain/application/Application.java]: In all log statements, let's stick to using descriptive language. I.e. 

```
throw new SecurityException(
                    "Requesting team member [{}] is not a valid member of any team associated to the owning project, {}",
                        command.getRequestedBy().getTeamMemberId(),
                        projectId.getProjectId());
```

File: nucleus/src/main/java/za/co/absa/subatomic/domain/application/Application.java [hunk 4]
```diff
@@ -33,6 +34,12 @@
 
     @CommandHandler
     public Application(NewApplication command) {
+        if (!command.getAllAssociateProjectOwnerAndMemberIds()
+                .contains(command.getRequestedBy().getTeamMemberId())) {
+            throw new SecurityException(
```
Comment (donovanmuller) [inline nucleus/src/main/java/za/co/absa/subatomic/domain/application/Application.java]: Assuming you're using [`java.lang.SecurityException`](https://docs.oracle.com/javase/8/docs/api/java/lang/SecurityException.html).

I'd stay clear of pigging backing off Exceptions meant for internal use. Instead, create a meaningful Gluon exception like `ApplicationAuthorisationException` that extends `RuntimeException`. You can also use Spring MVC's [`@ExceptionHandler`](http://www.baeldung.com/exception-handling-for-rest-with-spring#controlleradvice) to handle the exception and assign the correct HTTP response codes in the REST adapter.

Comment (kbristow) [reply nucleus/src/main/java/za/co/absa/subatomic/domain/application/Application.java]: I even spoke to you about doing this. Then I just flat out forgot to... :) oops

File: nucleus/src/main/java/za/co/absa/subatomic/domain/project/NewProject.java (hunk 1 above)
Comment (kbristow) [reply nucleus/src/main/java/za/co/absa/subatomic/domain/project/NewProject.java]: Mistake
